packages/grid/x-grid/src/tests/filtering.XGrid.test.tsx

```diff
@@ -285,12 +280,12 @@ describe('<XGrid /> - Filter', () => {
       />,
     );
 
+    expect(handleFilterChange.callCount).to.equal(1);
     expect(getColumnValues()).to.deep.equal(['Adidas', 'Puma']);
     const input = screen.getByPlaceholderText('Filter value');
     fireEvent.change(input, { target: { value: 'ad' } });
     clock.tick(SUBMIT_FILTER_STROKE_TIME);
-    expect(visibleRows.size).to.equal(1);
-    expect(visibleRows.get(1)).to.deep.equal({ id: 1, brand: 'Adidas' });
```

Comment:
I would argue that this assertion was wrong. Changing the filter model doesn't mean the rows are filtered.